lib/main.go

```diff
@@ -214,7 +215,29 @@ func (obj *Main) Run() error {
 		return fmt.Errorf("hostname cannot be empty")
 	}
 
+	// Get current user running mgmt
+	user, err := user.Current()
+	if err != nil {
+		return errwrap.Wrapf(err, "can't get current user")
+	}
+
+	// Use systemd SateDirectory if set. If not, use
+	// XDG_CACHE_DIR unless user is root, then use
+	// /var/lib/mgmt
 	var prefix = fmt.Sprintf("/var/lib/%s/", obj.Program) // default prefix
+	stateDir := os.Getenv("STATE_DIRECTORY")
+	xdg := os.Getenv("XDG_CACHE_HOME")
+
+	if xdg == "" {
+		xdg = fmt.Sprintf("%s/.cache/%s", user.HomeDir, obj.Program)
+	}
+
+	if stateDir != "" {
+		prefix = stateDir
+	} else if user.Uid != "0" {
+		prefix = fmt.Sprintf("%s", xdg)
+	}
```

Comment:
Probably need to add a check to add trailing slash if one is not present. (STATEDIR or XDG might not necessarily provide one.)

Reply:
I tested by both using StateDirectory in the unit file and STATE_DIRECTORY as an environment variable without it. I didn't pass the / to either and it put it under each correctly.

Reply:
It's likely IIRC that later parts of the code that consume `prefix` check for it and add one, but we should do it anyways in case they don't.

Reply:
ok will do

Reply:
So if I have it add the / suffix, it errors out here:
```
pgpPrefix := fmt.Sprintf("%s/", path.Join(prefix, "pgp"))
                if err := os.MkdirAll(pgpPrefix, 0770); err != nil {
                        return errwrap.Wrapf(err, "can't create pgp prefix")
                }
```

Looks like since the / is in the path there, it's adding a double slash and you get the error:

    21:03:28 run: error: can't create pgp prefix: mkdir /pgp/: permission denied

Reply:
So I realized it doesn't matter anyway. I need to add `obj.Program` to the path. Because the mgmt  directory should exist in the XDG_CACHE_HOME directory, so the slash will be added to prefix when we assign xdg.

Reply:
Just ignore me, I'm dumb. It works with adding the trailing slash.